source/secure_tunneling.c

```diff
@@ -308,6 +361,7 @@ static int s_aws_secure_tunnel_set_stream(
     if (service_id == NULL || service_id->len == 0) {
         secure_tunnel->connections->stream_id = stream_id;
         aws_hash_table_clear(&secure_tunnel->connections->connection_ids);
+        /* TODO This case should be treated as error. */
```

Comment:
Why? I think it was allowed to have a connection id with no service_id. I believe the case was when someone opens a secure tunnel with no service ids but you can still connect via v3 protocol to open as many v3 connection_id streams as you'd like. The protocol was a bit vague on it but I think I tested and it was allowed? I could be wrong though, it's been a while.

Reply:
For history. The docs don't allow specifying connection id for en empty service id, though in a really vague way. But the security tunnel service allows it. And our library supports this functionality. So, we let it be, at least for now.